### ISSUE TITLE
NVDA crashes with a specific SAPI5 synthesizer

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3986]
**Reported by k_kolev1985 on 2014-03-16 18:30**
I've recently upgraded my machine to Windows 8.1 64-bit. I've installed a few SAPI5 synthesizers: eSpeak (32-bit), SpeechLab (a 32-bit bulgarian speech synthesizer) and Irina (another 64-bit compatible commercial bulgarian speech synthesizer, developed by the greek company "Innoetics"). I've created a few profiles in NVDA, so I can easily switch between those voices/synthesizers. I have profiles for SpeechLab, for MS David (witch should be 64-bit compatible) and for Irina. When I switch to the "Irina" profile and then try to switch to the normal configuration however, NVDA crashes. No error sounds, no dialogs - just a silent crash. The crash occurs as soon as I press Enter on the currently active profile in the "Configuration profiles" dialog in order to switch from it to the normal configuration. The synthesizer "Irina" is a 64-bit compatible one - it installed itself in "Program Files" folder and it appears in Control Panel -> Text-To-Speech and can be used in Narrator. The crash occurs even if you switch from "Irina" to another synthesizer via the "Synthesizer" dialog. I've managed to procure a log file (there was no .dmp file) and I'm attaching it - hope it helps. Note, that I am not experiencing this problem with the same synthesizers under Windows 7 Ultimate SP1 32-bit and Windows 8.1 32-bit - only under the Windows 8.1 Pro N 64-bit machine.

My system specifications: 
- OS: Windows 8.1 Pro N, 64-bit, in bulgarian. 
- NVDA version: next-10434,4c80d23; 2014.1.

**Comment by k-kolev1985** [comment]
Update: The crash described above occurs with other 3rd party SAPI5 TTS engines (e.g. SAPI5 Eloquence from Code Factory) as well, and not only with our bulgarian voice "Irina". But it seams not to occur with the built-in SAPI5 TTS engines from Microsoft. It seams to occur only with 3rd party SAPI5 TTS engines witch support the 64-bit architecture. If needs be, I'll provide another log file from NVDA.

Test environment:
- Operating system: Windows 10 Pro (build 10586.11), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: next-12862,0367e74.
- Processor: Intel Core i5-2320 at 3.00GHz.
- RAM Memory: 4.00 GB.
- Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 2047MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC662 at Intel Cougar Point PCH - High Definition Audio Controller.

**Comment by jcsteh** [comment]
I have Ivona Kendra SAPI5 here which is 64-bit compatible and it doesn't occur with that, so it must be just specific synthesisers. Alternatively, there might be something weird with the SAPI5 runtime on your system, but that seems quite unlikely.

Unfortunately, log files aren't going to help here, since NVDA crashes before it can log anything. I have a horrible suspicion the synthesiser is choosing to exit the process for some reason. I know that Code Factory have some pretty wacky copy protection code and it's possible it thinks there has been a copy protection violation of some sort (regardless of whether there actually has been).

**Comment by k-kolev1985** [comment]
I doubt that I have a broken SAPI5, because it seams to work OK. And this issue has persisted between several reinstalls of Windows 8.1 and 10.

Strangely enough, I can't remember having this issue on Windows 7, but then it was a 32-bit and not a 64-bit one as it is now.

Your guess about the copy protection may be right. The voice "Irina" is also a commercial one and probably also has a copy protection as does Eloquence. And I remember that from time to time it checks the internet for its license, as does Eloquence from Code Factory - maybe it has something to do with that (?).

So, maybe it is a good idea to contact the developers of "Irina" (the greek company "Innoetics") so eventually the problem with "Irina" could be solved (?). I can do that if needs be. Would it be OK if I give them some contact information of NVAccess, so they can contact you if necessary?

**Comment by zahyur** [comment]
Hi,
I can reproduce this under Windows 10 Insider preview 64-bit with a demo of Innoetics Irina SAPI5 synthesizer.
The difference between the demo and a licensed version is a background music.
You can download it from this page:
http://www.innoetics.com/index.php?option=com_weblinks&view=category&id=46&Itemid=301&lang=en
Direct link:
http://archive.spellcastcloud.com/TtSReaderBGI.zip

1. Install the synthesizer.
Extract the archive and start the executable. When it asks for the additional files, point it to the directory you've extracted the archive.
If the installer is in Bulgarian, just press the Next button on every page, the one which has a greater sign at the end.
On the step, where you have to accept the license agreement, there are two radio buttons and the default is "Do not accept" - select the other one.
On the last step, there is a check box, which starts the activator after finishing - uncheck that.
2. After installing, select SAPI5 from the NVDA's Synthesizer dialog.
3. Change the voice to Irina (don't worry, it speaks English as well).
4. Change the synthesizer to ESpeak.
5. Boom! NVDA crashes.

If you change the voice to another SAPI5 voice and then change it to ESpeak, NVDA crashes regardless. It seems that as soon as Irina is used in the current session, you can't switch away from SAPI5 anymore.

Kosta, if you decide to write to Innoetics, point them to this issue directly. And don't worry - they know NVDA very well - they've promoted it way before Irina was officially released.

**Comment by k-kolev1985** [comment]
OK, after the last changes to the SAPI5 support of NVDA ("next" snapshots) , the issue started occurring with it seams all (incl. only 32-bit compatible) SAPI5 voices. I have Microsoft Zira and it occurs with it as well. Previously it did occur with only voices that are 64-bit compatible.

Test environment:
- Operating system: Windows 10 Pro (build 10586.420), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: next-13421,dd10cdbe, in bulgarian.
- Processor: Intel Core i5-2320 at 3.00GHz.
- RAM Memory: 8.00GB.
- Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 3071MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC662 at Intel Cougar Point PCH - High Definition Audio Controller.

**Comment by michaelDCurran** [comment]
I don't understand. NVDA never had support for 64-bit voices. Can you 
explain further?


Do you see the crash with Microsoft Speech Mobile voices (if you have 
them in that build of Windows 10)?


For any crashes, please provide an NVDA log and crash dump if possible.





On 13/07/2016 7:24 PM, Kostadin Kolev wrote:
>
> OK, after the last changes to the SAPI5 support of NVDA ("next" 
> snapshots) , the issue started occurring with it seams all (incl. only 
> 32-bit compatible) SAPI5 voices. I have Microsoft Zira and it occurs 
> with it as well. Previously it did occur with only voices that are 
> 64-bit compatible.
>
> Test environment:
>
>   * Operating system: Windows 10 Pro (build 10586.420), 64-bit, in
>     Bulgarian with all locale settings set to "Bulgarian".
>   * NVDA version: next-13421,dd10cdbe, in bulgarian.
>   * Processor: Intel Core i5-2320 at 3.00GHz.
>   * RAM Memory: 8.00GB.
>   * Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 3071MB
>     shared system memory, desktop resolution set to 1920x1080.
>   * Sound Card: Realtek ALC662 at Intel Cougar Point PCH - High
>     Definition Audio Controller.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/3986#issuecomment-232303475>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ANf5nfsMMzgcDpj8cGZjrqmYngQaThpFks5qVK6zgaJpZM4GsfuS>.
>

-- 
Michael Curran
Executive Director, NV Access Limited
Phone: +61 7 3149 3306
Website: http://www.nvaccess.org/
Twitter: @NVAccess
Facebook: http://www.facebook.com/NVAccess

**Comment by k-kolev1985** [comment]
The issue started occurring after I switched to a 64-bit Windows 8.1. The issue was occurring only with the voices that had a 64-bit variant installed as well, even though NVDA was using their 32-bit variant.

After the last changes to NVDA in "next" snapshots however, the crashes started occurring with seams it all SAPI5 voices. At the moment I can't confirm if it occurs with the mobile voices as well, because my Windows 10 Insider Preview virtual machine is updating to the latest test build of Windows 10 and my physical machine is running a Stable version of Windows 10 in witch the mobile voices are not present.

I've managed to again reproduce the issue with the log level set to "Input/Output" with all add-ons disabled. I'll attach the log here. I did not find a crash DMP file however.
[nvda-old.txt](https://github.com/nvaccess/nvda/files/363318/nvda-old.txt)

**Comment by k-kolev1985** [comment]
OK, thanks.

And just for reference, I can't test this issue on my Windows 10 Insider Preview virtual machine, because it is a 32-bit one and the issue seams to be occurring only on 64-bit machines. I've tried it just to make sure and for sure - I can't reproduce the issue there.

**Comment by bhavyashah** [comment]
@michaelDCurran You noted in https://github.com/nvaccess/nvda/issues/3986#issuecomment-232620249 that you would resume work on this when you had more time. Since that has evidently not happened due to other priorities, and since assigning priorities (P1, P2, P3 and P4) is something that I believe we have started fairly recently, feel free to assign a priority you deem appropriate for this one also.

**Comment by Adriani90** [comment]
the referenced pull request has been closed and the issue seems to be handled in pull request #8174
@k-kolev1985 can you still reproduce it?

**Comment by k-kolev1985** [comment]
@Adriani90 I've tried to reproduce this now, but I wasn't able to. Maybe it got fixed in NVDA at some point - I don't know.

**Comment by Adriani90** [comment]
Ok, then maybe we can close it and reopen if the issue comes up again. Unfortunately I cannot close it.

 

 

 

Von: Kostadin Kolev <notifications@github.com> 
Gesendet: Sonntag, 23. Dezember 2018 22:11
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Mention <mention@noreply.github.com>
Betreff: Re: [nvaccess/nvda] NVDA crashes with a specific SAPI5 synthesizer (#3986)

 

@Adriani90 <https://github.com/Adriani90>  I've tried to reproduce this now, but I wasn't able to. Maybe it got fixed in NVDA at some point - I don't know.

**Comment by ehollig** [comment]
Closing
